### ISSUE TITLE
Put scripts in footer

### ISSUE THREAD

**Issue by Froal** [deconf/Google-Analytics-Dashboard-for-WP#2]
Is that possible?
```php
if (! class_exists('GADASH_Tracking')) {

    class GADASH_Tracking
    {

        function __construct()
        {
            add_action('wp_footer', array(
                $this,
                'ga_dash_tracking',
            ), 999);
            add_action('wp_enqueue_scripts', array(
                $this,
                'ga_dash_enqueue_scripts'
            ));
        }
[...]
```
In tracking.php

**Comment by deconf** [comment]
It is, but there are some issues like:
- the user won't be able to verify its site using the Google Analytics code in Webmaster Tools if the tracking code is not inserted in <head> section (https://support.google.com/webmasters/answer/185871?hl=en)
- wp-footer action is missing for certain themes
- having the code in footer may also affect your analytics data accuracy

The code is asynchronous, are there any particular reasons for having it in footer?

**Comment by woozat** [comment]
Hello, The tracking code appears — visually I mean — in every page of my site, also modifying the position of the title of the page ... 

My page : 

www.philtaka.fr

1- Is it possible to make the text font size very tiny as it (almost) would not interfere graphically so much ?
2- And/or make it invisible ...

Regards, 

Phil

**Comment by Froal** [comment]
@deconf Is it possible to just make it as an option in the backend, then?
Users who want to use it in the footer won't have problem, that way.

Particular reason: some sites want to have all javascript in footer, including jquery - at the end of the day it's not so much an issue with having GADASH_Tracking's code in the footer, but more its assumption that jQuery will be there that is an issue.

**Comment by deconf** [comment]
FYI: That code is inserted by your theme, it has nothing to do with this plugin. This is a support question, so please use https://deconf.com/ask/ for further postings.

**Comment by Froal** [comment]
@woozat it is a different issue... Please do not hijack existing current discussions.

**Comment by woozat** [comment]
Ok, sorry ...

**Comment by deconf** [comment]
I think I've tested that in the past in one branch and if I remember well there was an issue while enqueuing scripts in the footer. I was unable to insert the event tracking code after jQuery resulting in a JavaScript error.

The presence of jQuery is not required, the plugin automatically enqueues it only when needed (only if event tracking is used to be more accurate).

**Comment by deconf** [comment]
https://github.com/deconf/Google-Analytics-Dashboard-for-WP/blob/master/front/tracking.php#L29

**Comment by Froal** [comment]
If scripts are all in footer (including jquery) via this code in function.php:
```php
add_action('wp_enqueue_scripts', 'theme_clean_head');
function theme_clean_head() {
	if(!is_admin()) {
		remove_action('wp_head', 'wp_print_scripts'); 
		remove_action('wp_head', 'wp_print_head_scripts', 9); 
		remove_action('wp_head', 'wp_enqueue_scripts', 1); 
	}
}
```
Or by using https://wordpress.org/plugins/scripts-to-footerphp (it does the same!)

We get:
```
Uncaught ReferenceError: jQuery is not defined
```

**Comment by caleb531** [comment]
@deconf—I use your wonderful plugin and have been interested in reordering the inline analytics script as well. However, I understand that placing the script in the footer is likely to cause issues. My issue is slightly different.

### TL;DR

I'd like an option to force the Google Analytics inline script to be inserted at the end of the `<head>`

### Explanation

I run all of the sites I manage through [GTMetrix](http://gtmetrix.com/), which rates my site's performance via the PageSpeed and YSlow benchmarks. According to the PageSpeed results (which GTMetrix provides) upon checking my site, I am notified that my site includes one inline script blocking the rendering of my page. A quick check of the page source reveals that the Google Analytics code has been inserted *between* two `<link>` elements. To solve this, I would recommend forcing the inline script to be at the end of the `<head>` (this can probably be assigning to the `wp_enque_script` action a very high priority level.

**Comment by deconf** [comment]
wp_enque_script doesn't apply here. Lowering the priority while hooking on wp_head works, see: 

https://github.com/deconf/Google-Analytics-Dashboard-for-WP/commit/c1d514f96a1210bce4b86bd0b7868ba6bef5337a

**Comment by caleb531** [comment]
I meant to say `wp_head`, not `wp_enqueue_scripts`; I don't know what I was thinking. Anyway, what you've done should work perfectly. 

Thanks very much,
Caleb

**Comment by deconf** [comment]
@Froal, on newer versions (> 4.5) you'll be able to move the script as you wish like this:

https://gist.github.com/deconf/ad5fa23ec3de4eac517c

FYI: That's why I've closed this ticket.

**Comment by Froal** [comment]
Great, thank you!
